NETCORE/src/Shared/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -158,14 +158,25 @@ public static IServiceCollection ConfigureTelemetryModule<T>(this IServiceCollec
         /// <param name="developerMode">Enables or disables developer mode.</param>
         /// <param name="endpointAddress">Sets telemetry endpoint address.</param>
         /// <param name="instrumentationKey">Sets instrumentation key.</param>
-        /// <param name="connectionString">Sets connection string.</param>
         /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddApplicationInsightsSettings(this IConfigurationBuilder configurationSourceRoot,  bool? developerMode = null, string endpointAddress = null, string instrumentationKey = null)
+            => configurationSourceRoot.AddApplicationInsightsSettings(developerMode, endpointAddress, instrumentationKey);
```

Comment:
This method call does not seem to use `connectionString` parameter or assign it to null - can you take a look?

Reply:
wow, so I was blindly trusting unit tests to catch something like this.
what actually happened was that it is an infinite loop, which crashes the test assembly and no errors get reported. 
learning lots of things today